components/ws-manager/pkg/manager/monitor.go

```diff
@@ -800,7 +800,8 @@ func (m *Monitor) finalizeWorkspaceContent(ctx context.Context, wso *workspaceOb
 		tpe = api.WorkspaceType_REGULAR
 	}
 
-	doBackup := wso.WasEverReady() && !wso.IsWorkspaceHeadless()
+	doBackup := wso.WasEverReady() && !wso.IsWorkspaceHeadless() && !wsk8s.IsGhostWorkspace(wso.Pod)
```

Comment:
if `wsk8s.IsGhostWorkspace(wso.Pod) == true` then `wso.IsWorkspaceHeadless() == true`. 


If that's not the case, we have a bug somewhere else.